fix(regen-defs): Fix globs and clean tempfiles on INT, TERM

aaronliu0130 opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fixes**
  * Improved interruption handling so temporary output files are reliably cleaned up when the generation/linting process is stopped.

* **Refactor**
  * Centralized temporary-file cleanup into a single routine and aligned command execution with the updated cleanup flow.
  * Updated the configured lint tool path to use the correct project directory.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

aaronliu0130 committed:
fix(regen-defs): Fix globs and clean tempfiles on INT, TERM

cclauss commented on regen-defs.zsh:
```diff
@@ -23,8 +23,14 @@ for folder in $folders; do
+
     # Execute the command and capture stdout and stderr
-    uv run "$cpplint" $cmd > "$stdout_file" 2> "$stderr_file"
+    eval uv run "$cpplint" $cmd > "$stdout_file" 2> "$stderr_file"
```
How does this change things?

aaronliu0130 replied:
It makes globbing work, since otherwise we literally pass "*" to cpplint, at least on my machine.

aaronliu0130 commented on regen-defs.zsh:
```diff
@@ -23,8 +23,14 @@ for folder in $folders; do
     stdout_file=$(mktemp)
     stderr_file=$(mktemp)
 
+    # Cleanup on interruption
+    cleanup() {
+      rm "$stdout_file" "$stderr_file"
+    }
+    trap cleanup INT TERM
```
@copilot By the time the trap is re-set, wouldn't the previous values already be cleaned?

aaronliu0130 committed:
use wildcard parameter expansion instead of eval

I do not think "attackers controlling .def content" is a plausible scenario, but this does seem to be the better tool for the job.

aaronliu0130 commented on regen-defs.zsh:
```diff
@@ -42,7 +48,7 @@ for folder in $folders; do
 
     # Clean up temporary files
-    rm "$stdout_file" "$stderr_file"
+    cleanup
```
I don't think a command in between can cause termination of the entire script, and as mentioned trapping EXIT can get a bit messy.

aaronliu0130 commented on regen-defs.zsh:
```diff
@@ -23,8 +23,14 @@ for folder in $folders; do
     stdout_file=$(mktemp)
     stderr_file=$(mktemp)
 
+    # Cleanup on interruption
+    cleanup() {
+      rm "$stdout_file" "$stderr_file"
+    }
+    trap cleanup INT TERM
```
Does that mean I should add the EXIT trap or not?

aaronliu0130 committed:
back to eval—switch to expansion caused all kinds of breakage

lovewave02 reviewed (changes requested):
The signal handler cleans the temp files but does not terminate the script. In zsh, a handled `INT` or `TERM` resumes after the trap; a minimal reproduction prints `CLEANED` and then `CONTINUED_AFTER_INT`.

Here that means `regen-defs.zsh` can continue after Ctrl-C and rewrite the current `.def` file from partial output. Please have the handler clean up and exit with the matching signal status, and add a small regression check for the interrupted path.

aaronliu0130 committed:
fix: make trap exit

aaronliu0130 committed:
fix: change default cpplint directory to /Projects/cpplint

see https://blog.tenstral.net/2026/04/hello-projects-directory.html .
change mostly because this is what i now use

androvonx95 reviewed (changes requested):
Nice to see the trap actually terminating now, but I think `53a1696` has a side effect: `exit` went into `cleanup`, and `cleanup` is also what replaced the `rm` at the end of each loop iteration.

```sh
# Clean up temporary files
cleanup      # <- ends in `exit`, so the success path exits too
```

So the run stops after the first `.def` file instead of continuing through the rest.

Also, `trap cleanup INT TERM` invokes the function with no arguments — `$1` is only set in the `TRAPINT()`-style form — so `exit $((128 + $1))` isn't valid arithmetic on either path.

Splitting the two responsibilities keeps the loop intact and still satisfies the signal-status request:

```sh
cleanup() { rm -f "$stdout_file" "$stderr_file"; }
trap 'cleanup; exit 130' INT
trap 'cleanup; exit 143' TERM
```

With the end-of-iteration call left as plain `cleanup`. I verified this control flow in bash (loop completes; `INT` exits 130) — I don't have the environment here, so you'll want to confirm on your side.

Unrelated to the title: `e89746c` swaps `$HOME/Documents/cpplint` for `$HOME/Projects/cpplint`, which just trades one local layout for another. Deriving it would work for everyone:

```sh
cpplint="${0:A:h}/cpplint.py"
```

(set before the `cd samples/`).

PNHD reviewed (changes requested):
Current head `e89746c` still has two merge blockers.

`cleanup()` now exits, but the normal success path calls `cleanup` at the end of every iteration, so the script stops after the first `.def`. The INT/TERM trap also calls `cleanup` without a signal-number argument, so `$1` is not valid for the exit-status arithmetic.

Separately, `eval` reparses the command and can split a quoted cpplint path containing whitespace.

Please separate ordinary cleanup from signal-exit handling and preserve the cpplint path as one shell word.
